### PR TITLE
Add a standalone parallel iterator to the leader/follower primer and rename it

### PR DESCRIPTION
**Pull request by daviditen**
Add a standalone parallel iterator to leaderfollower.chpl that will be used in
the two non-zippered foralls.  Renamed this primer to parIters.chpl. Updated
references to this test use the new name in two READMEs, a Makefile, and the
iterators.chpl primer.

### CODE REVIEW

File: test/release/examples/primers/iterators.chpl [hunk 1]
```diff
@@ -142,10 +142,10 @@ writeln();
...
 // iterators that create parallel tasks and assign work to
 // those tasks are required.  These are known as leader/follower
-// iterators in Chapel, and they are a fairly big topic of their
-// own.  See the leaderfollower.chpl test in this directory for
-// an extended example of defining and using leader/follower
-// iterators.
+// or standalone paralllel iterators in Chapel, and they are a fairly
+// big topic of their own.  See the parIters.chpl test in
+// this directory for an extended example of defining and using
+// leader/follower and standalone parallel iterators.
```
Comment (lydia-duncan) [inline test/release/examples/primers/iterators.chpl]: I'd be tempted to change the last two sentences of this paragraph to just "Such iterators are a fairly big topic and are described in detail in the parIters.chpl test which lives in this directory", on the argument that the different types of parallel iterators would be covered in there anyways.

If you chose to leave it as is, there's an extra letter "L" in parallel.

**Comment by lydia-duncan** [comment]
Looks great, thanks David!

**Comment by daviditen** [comment]
Thanks Lydia.  I made a slight change to your suggested wording in the comment in iterators.chpl, but mostly used your suggestion.

**Comment by lydia-duncan** [comment]
I like it :)

**Pull request merged**